ApeWorX/ape
# fix: better warnings for missing compilers

### What I did

The warning for missing compilers needed an iteration... it told me compilers for markdown files were missing, which was weird. Also, I would have liked an additional hint telling me possibly one of the compiler plugins did not load correctly.

### How I did it

* Filter out common extensions like `.md` and `.adoc` from the warning.
* Handle plurality and singularity independently
* Attach suffix for indicating possible failed to load plugins
* Update one of the existing tests to include a check for all of this

### How to verify it

No longer see unnecessary warning about missing compilers for `.md` files and `.adoc` files and stuff
The warning you do you get when something like `ape-vyper` failing to load is a little better. Still if we could get a handle on https://github.com/ApeWorX/ape/issues/614 it could be even better yet.

### Checklist
